crates/winch/src/compiler/mod.rs

```diff
@@ -0,0 +1,85 @@
+use crate::isa::TargetIsa;
+use anyhow::Result;
+use wasmtime_environ::{
+    CompileError, DefinedFuncIndex, FunctionBodyData, ModuleTranslation, ModuleTypes, Tunables,
+};
+
+pub mod builder;
+
+pub(crate) struct Compiler {
+    isa: Box<dyn TargetIsa>,
+}
+
+impl Compiler {
+    pub fn new(isa: Box<dyn TargetIsa>) -> Self {
+        Self { isa }
+    }
+}
+
+impl wasmtime_environ::Compiler for Compiler {
```

Comment:
If we want to break the dependency on Wasmtime, the way to do that would be to have a `crates/winch` crate analogous to `crates/cranelift` (`wasmtime-cranelift`) that pulls in the actual Winch toplevel crate (just `winch`, or `winch-codegen` or somesuch) and impls this Wasmtime trait. We can always refactor in that direction later, but if you wanted to do it now I think that could be reasonable as well.

Reply:
I've tackled this too, I think that it's better to do this earlier rather than later.